includes/blocks/class-sensei-continue-course-block.php

```diff
@@ -63,8 +63,13 @@ public function render( array $attributes, string $content ) : string {
 		}
 
 		$target_post_id = Sensei_Utils::get_target_page_post_id_for_continue_url( $course_id, $user_id );
+		$target_url     = get_permalink( absint( $target_post_id ?? $course_id ) );
+		if ( ! $target_url ) {
+			return '';
+		}
 
-		return '<form action="' . esc_url( get_permalink( absint( $target_post_id ?? $course_id ) ) ) . '" method="get" class="sensei-block-wrapper sensei-cta">' .
+		return '<form action="' . esc_url( $target_url ) . '" method="get" class="sensei-block-wrapper sensei-cta">' .
+			Sensei_Utils::output_query_params_as_inputs( [], $target_url, false ) .
```

Comment:
Just a thought I wanted to share, not a preference or recommendation. As this function is no longer just outputting, it also returns the content without exactly 'outputting', should we rename it to something else?

Reply:
I'm not in love with this method as well but I think it's ok because outputting is the default behavior. Looking at the codebase, it's mainly used in output mode.